lib/Dialect/TritonGPU/Transforms/Coalesce.cpp

```diff
@@ -31,7 +37,8 @@ struct CoalescePass : public TritonGPUCoalesceBase<CoalescePass> {
     unsigned maxContig = info.getContiguity(order[0]);
     unsigned alignment = std::min(maxMultiple, maxContig);
     unsigned perThread = std::min(alignment, 128 / numBits);
-    sizePerThread[order[0]] = perThread;
+    sizePerThread[order[0]] = std::min<int>(perThread, numElemsPerThread);
```

Comment:
In my understanding, I treat sizePerThread as `nts` in the original code, that's why I port the original [logic](https://github.com/openai/triton/blob/master/lib/codegen/analysis/layout.cc#L309) here.

Please correct me if I am wrong @ptillet

Reply:
Yep, `sizePerThread` and `nts` represent the same thing